week2/project/ecommerce/src/buttonSet.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState, useEffect } from 'react';
+import Button from './Button.jsx';
+
+const ButtonSet = ({ setFilterProducts }) => {
+    const [categories, setCategories] = useState([]);
+
+    const fetchCategories = async () => {
+        try {
+            const response = await fetch('https://fakestoreapi.com/products/categories');
+            const data = await response.json();
+            setCategories(data); 
+        } catch (error) {
+            console.error('Error fetching categories:', error);
+        }
+    };
+
+    useEffect(() => {
+        fetchCategories();
+    }, []);
+
+    return (
+        <div className="button-set">
+            {categories.map((category, index) => (
+                <Button
+                    key={index}
+                    buttonName={category}
+                    setFilterProducts={setFilterProducts}
```

Comment:
```suggestion
                    onFilterProducts={setFilterProducts}
```
Suggestion:
For general naming in React, props that handle events should start with 'on'.

The 'set' prefix is typically reserved for useState hook naming
The name should describe what the prop does, not how it does it
[Reference](https://react.dev/learn/responding-to-events#naming-event-handler-props:~:text=By%20convention%2C%20event%20handler%20props%20should%20start%20with%20on%2C%20followed%20by%20a%20capital%20letter.)

Reply:
Now the button is represented differently in this file. 
setFilterProducts is parameter from useState hook from App.jsx.